pkg/cloud/ibm/assets/deployment.yaml

```diff
@@ -0,0 +1,114 @@
+apiVersion: apps/v1
+metadata:
+  name: ibm-cloud-controller-manager
+  namespace: openshift-cloud-controller-manager
+  labels:
+    k8s-app: ibm-cloud-controller-manager
+spec:
+  replicas: 3
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 1
+      maxSurge: 3
+  selector:
+    matchLabels:
+      k8s-app: ibm-cloud-controller-manager
+  template:
+    metadata:
+      labels:
+        k8s-app: ibm-cloud-controller-manager
+    spec:
+      priorityClassName: system-cluster-critical
+      hostNetwork: true
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - topologyKey: "kubernetes.io/hostname"
+            labelSelector:
+              matchLabels:
+                k8s-app: ibm-cloud-controller-manager
+      tolerations:
+      - effect: NoSchedule
+        key: node-role.kubernetes.io/master
+        operator: Exists
+      - effect: NoExecute
+        key: node.kubernetes.io/unreachable
+        operator: Exists
+        tolerationSeconds: 120
+      - effect: NoExecute
+        key: node.kubernetes.io/not-ready
+        operator: Exists
+        tolerationSeconds: 120
+      - effect: NoSchedule
+        key: node.cloudprovider.kubernetes.io/uninitialized
+        operator: Exists
+      - effect: NoSchedule
+        key: node.kubernetes.io/not-ready
+        operator: Exists
+      serviceAccountName: cloud-controller-manager
+      terminationGracePeriodSeconds: 90
+      containers:
+      # Container is required to specify name in order for image substitution
+      # to take place. Cloud-controller-manager is substituted with
+      # real CCM image from the cluster
+      - name: cloud-controller-manager
+        image: quay.io/openshift/origin-ibm-cloud-controller-manager
+        imagePullPolicy: IfNotPresent
+        securityContext:
+          runAsGroup: 0
+          runAsNonRoot: true
+        env:
+          - name: VPCCTL_CLOUD_CONFIG
+            value: /etc/kubernetes/cloud.conf
+        command:
+          - /bin/bash
+          - -c
+          - |
+            #!/bin/bash
+            set -o allexport
+            if [[ -f /etc/kubernetes/apiserver-url.env ]]; then
+              source /etc/kubernetes/apiserver-url.env
+            fi
+            exec /bin/ibm-cloud-controller-manager \
+            --port=0 \
+            --use-service-account-credentials=true \
+            --kubeconfig=/etc/kubernetes/kubeconfig \
+            --authentication-kubeconfig=/etc/kubernetes/kubeconfig \
+            --authorization-kubeconfig=/etc/kubernetes/kubeconfig \
+            --client-ca-file=/etc/kubernetes/ca.crt \
+            --requestheader-client-ca-file=/etc/kubernetes/ca.crt \
+            --cloud-provider=ibm \
+            --cloud-config=/etc/kubernetes/cloud.conf \
+            --feature-gates=LegacyNodeRoleBehavior=false,IPv6DualStack=false \
```

Comment:
`LegacyNodeRoleBehavior=false` is no longer a feature gate for our 1.22 version of the CCM.  Please remove it.

Reply:
updated